### DISCUSSION TITLE
Error handling non 500 errors

### DISCUSSION THREAD

**Discussion by danieljee** [vercel/next.js#54595] [App Router]
Next.JS Version = 13.4.13

I'm wondering if there is a way for error.tsx to differentiate different types of server-side errors.

For example, if I have a server side component like this:

```
const Layout = async ({  children }) => {
  const { data, error } = await api.getStuff();

  if (error) {
    if (error.statusCode === 404) return notFound();
    throw error;
  }

  return (
    <div>{children}</div>
  )
}
```
If the api's response is 404, I can display not-found.tsx to the end user. However for all other errors, I'm displaying error.tsx which only shows 500 internal error. 

Is there a way to display different status code + message to the end user?

**Comment by alxhotel** [comment]
We need a way to differentiate between different types of errors.

This "security feature" is **not useful**

![image](https://github.com/vercel/next.js/assets/1944024/ce81236c-1af2-49a7-8527-61bef6498411)
